docker/Dockerfile

```diff
@@ -57,20 +58,72 @@ ENV LD_LIBRARY_PATH "/usr/local/lib:$LD_LIBRARY_PATH"
 # install KINC
 WORKDIR /opt
 
-ADD https://api.github.com/repos/SystemsGenetics/KINC/git/refs/heads/${KINC_REVISION} kinc-version.json
-RUN git clone -q https://github.com/SystemsGenetics/KINC.git \
-	&& mkdir -p KINC/build \
-	&& cd KINC/build \
-	&& git checkout -q ${KINC_REVISION} \
-	&& qmake ../src/KINC.pro GUI=no \
-	&& make -s -j $(nproc) \
-	&& make -s qmake_all \
-	&& make -s install
+# For older version of KINC we need to use qmake. For newer versions after 3.3.x we
+# can use the make file in the root directory of KINC.
+RUN if [ "${KINC_REVISION}" = "v3.3.x" ] || [ "${KINC_REVISION}" = "v3.2.x" ] ; then \
+        git clone -q https://github.com/SystemsGenetics/KINC.git \
+        && cd KINC/build \
+    	&& git checkout-q ${KINC_REVISION} \
+    	&& qmake ../src/KINC.pro \
+    	&& make -s -j $(nproc) \
+    	&& make -s qmake_all \
+    	&& make -s install; \
+    else \
+        git clone -q https://github.com/SystemsGenetics/KINC.git \
+        && cd KINC \
+    	&& git checkout -q ${KINC_REVISION} \
+    	&& make -s -j $(nproc) \
+    	&& make -s install; \
+    fi
+
+
+# Add in a few additional requirements for the >=3.4 version of KINC. These are needed
+# for the R and Python scripts in the bin folder.
+RUN if [ ! "${KINC_REVISION}" = "v3.3.x" ] && [ ! "${KINC_REVISION}" = "v3.2.x" ] ; then \
+    apt-get install -qq -y \
+        m4 \
+        autoconf \
+        automake \
+        libxml2-dev \
+        bison \
+        flex \
+        build-essential \
+        libcurl4-gnutls-dev \
+        libssl-dev \
+        software-properties-common; \
+    fi
+
+# Install KINC.R
+ENV DEBIAN_FRONTEND=noninteractive
+RUN if [ ! "${KINC_REVISION}" = "v3.3.x" ] && [ ! "${KINC_REVISION}" = "v3.2.x" ] ; then \
+        apt-key adv --keyserver keyserver.ubuntu.com --recv-keys E298A3A825C0D65DFD57CBB651716619E084DAB9 \
+        && add-apt-repository 'deb https://cloud.r-project.org/bin/linux/ubuntu bionic-cran40/' \
+        && apt update \
+        && apt-get -qq install -y r-base \
+        && R -q -e "install.packages('devtools', dependencies=TRUE, repos='http://cran.us.r-project.org')"; \
+
+    fi
+
+RUN if [ ! "${KINC_REVISION}" = "v3.3.x" ] && [ ! "${KINC_REVISION}" = "v3.2.x" ] ; then \
+        R -e "library('devtools'); install_github('SystemsGenetics/KINC.R', ref='${KINC_R_REVISION}')"; \
+    fi
 
 # install python dependencies
 WORKDIR /opt/KINC
 
-RUN pip3 install -q -r requirements.txt
+RUN pip3 install  -r requirements.txt
+
+
 
 # initialize default work directory
 WORKDIR /workspace
+
+# Tini for signal processing and zombie killing.
+ENV TINI_VERSION v0.18.0
```

Comment:
Tini is no longer needed in docker builds, refer to Tini github repo: https://github.com/krallin/tini#using-tini

Reply:
I included it because I could not ctrl+c out of a docker image once it was running. After I added it I was able to do so.  I didn't realize I could just pass the `--init` flag to `docker run`. 
I don't see any harm in leaving it so we don't have to require the flag.